kyo-prelude/shared/src/test/scala/kyo2/CoroutineTest.scala

```diff
@@ -0,0 +1,255 @@
+        val effect =
+            for
+                _ <- Coroutine.spawn { a = 1 }
+                _ <- Coroutine.spawn { b = 2 }
+                _ <- Coroutine.spawn { c = 3 }
+            yield a + b + c
+
+        val result = Coroutine.run(effect)
+        assert(result.eval == 6)
+    }
+
+    "complex control flow" in run {
+        var executed = List.empty[Int]
+        def subRoutine(n: Int): Unit < Coroutine[Any] =
+            for
+                _ <- Coroutine.pause
+                _ <- Coroutine.spawn { executed = n :: executed }
+                _ <- if n > 0 then subRoutine(n - 1) else Coroutine.pause
+            yield ()
+
+        val effect =
+            for
+                _ <- Coroutine.spawn { executed = 1 :: executed }
+                _ <- subRoutine(3)
+                _ <- Coroutine.spawn { executed = 5 :: executed }
+            yield ()
+
+        val result = Coroutine.run(effect)
+        assert(result.eval == ())
+        assert(executed == List(5, 0, 1, 2, 3, 1))
+    }
+
+    "error propagation" in run {
+        val effect =
+            for
+                _ <- Coroutine.spawn { throw new RuntimeException("Spawn error") }
+                _ <- Coroutine.pause
+                _ <- Coroutine.spawn { () } // This should not execute
+            yield ()
+
+        assertThrows[RuntimeException] {
+            Coroutine.run(effect).eval
+        }
+    }
+
+    "large number of coroutines" in run {
+        val n   = 10000
+        var sum = 0
+        val effect =
+            for
+                _ <- Kyo.seq.collectUnit((1 to n).map(i =>
+                    Coroutine.spawn {
+                        sum += i
+                    }
+                ))
+            yield sum
+
+        val result = Coroutine.run(effect)
+        assert(result.eval == (1 to n).sum)
+    }
+
+    "Coroutine with Env and Var" in run {
+        val effect =
+            for
+                _ <- Coroutine.spawn {
+                    for
+                        env   <- Env.get[String]
+                        _     <- Var.update[Int](_ + 1)
+                        value <- Var.get[Int]
+                        _     <- Coroutine.pause
+                        _     <- Var.update[List[String]](s"$env-$value" :: _)
+                    yield ()
+                }
+                _ <- Coroutine.spawn {
+                    for
+                        env   <- Env.get[String]
+                        _     <- Var.update[Int](_ + 5)
+                        value <- Var.get[Int]
+                        _     <- Coroutine.pause
+                        _     <- Var.update[List[String]](s"$env-$value" :: _)
+                    yield ()
+                }
+            yield ()
+
+        val result =
+            Coroutine.run(effect)
+                .andThen(Var.get[List[String]])
+                .pipe(Var.run(List.empty[String]))
+                .pipe(Var.run(0))
+                .pipe(Env.run("env"))
```

Comment:
I forgot to mention in the initial prelude PR that I've introduced this `pipe` method. It seems convenient to make effect handling more fluent.